Dockerfile

```diff
@@ -1,10 +1,15 @@
-FROM zricethezav/gitleaks:v7.4.0
+FROM zricethezav/gitleaks:v8.7.2
 
 LABEL "com.github.actions.name"="gitleaks-action"
 LABEL "com.github.actions.description"="runs gitleaks on push and pull request events"
 LABEL "com.github.actions.icon"="shield"
 LABEL "com.github.actions.color"="purple"
 LABEL "repository"="https://github.com/zricethezav/gitleaks-action"
 
+# User 'root' is required to successfully modify the system `git` permissions so
+# that the GITHUB_WORKSPACE can be marked as safe, despite being owned by someone
+# else (i.e. a user outside of Docker).
+USER root
+
```

Comment:
This is imho not needed, as within the `entrypoint.sh` script the `$GITHUB_WORKSPACE` is defined as a safe directory. Further, one of the best practices while running containers is to run processes with a non-root user.

Reply:
Unfortunately the GITHUB_WORKSPACE directory cannot be marked as safe within the entrypoint without this change: in GitHub actions the system git config is on a volume mount that non-root users cannot write to.

I agree that it would be better not to do this though! Let me know if you have other ideas.

Reply:
One approach is to step down to the gitleaks user within entrypoint just before exec'ing gitleaks. I am not sure if this extra indirection would just complicate + add new potential error cases.